commands/src/main/java/org/wildfly/extras/creaper/commands/infinispan/cache/AddDistributedCache.java

```diff
@@ -0,0 +1,115 @@
+package org.wildfly.extras.creaper.commands.infinispan.cache;
+
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+/**
+ * @author Jan Martiska
+ */
+public class AddDistributedCache extends AbstractAddCache {
+
+    private final Boolean asyncMarshalling;
+    private final CacheMode mode;
+    private final Long queueFlushInterval;
+    private final Long remoteTimeout;
+    private final Double capacityFactor;
+    private final ConsistentHashStrategy consistentHashStrategy;
+    private final Long l1lifespan;
+    private final Integer owners;
+    private final Integer segments;
+
+    public AddDistributedCache(Builder builder) {
+        super(builder, CacheType.DISTRIBUTED_CACHE);
+        this.asyncMarshalling = builder.asyncMarshalling;
+        this.mode = builder.mode;
+        this.queueFlushInterval = builder.queueFlushInterval;
+        this.remoteTimeout = builder.remoteTimeout;
+        this.capacityFactor = builder.capacityFactor;
+        this.consistentHashStrategy = builder.consistentHashStrategy;
+        this.l1lifespan = builder.l1lifespan;
+        this.owners = builder.owners;
+        this.segments = builder.segments;
+    }
+
+    @Override
+    final Values addValuesSpecificForCacheType(Values generalCacheValues) {
+        return generalCacheValues
+                .andOptional("async-marshalling", asyncMarshalling)
+                .andOptional("mode", mode != null ? mode.getMode() : null)
+                .andOptional("queue-flush-interval", queueFlushInterval)
+                .andOptional("remote-timeout", remoteTimeout)
+                .andOptional("capacity-factor", capacityFactor)
+                .andOptional("consistent-hash-strategy",
+                        consistentHashStrategy != null ? consistentHashStrategy.getType() : null)
```

Comment:
I'm fine with skipping those parameters, if the builder methods have a javadoc stating something like "{@code capacity-factor} is only supported on WildFly 9 and above, this setting will be ignored on previous versions".